project4/name_server/bootstrap_driver.h

```diff
@@ -17,18 +16,20 @@ namespace nameserver {
  */
 class BootstrapDriver {
  public:
-
   /**
    * @param config_file The configuration file location
    */
-  explicit BootstrapDriver(const std::filesystem::path config_file);
+  explicit BootstrapDriver(std::filesystem::path config_file);
 
   /**
    * Start the user input loop
    */
   [[noreturn]] void Start();
 
  private:
+  /// Bootstrap
+  std::shared_ptr<nameserver::Bootstrap> bootstrap_;
```

Comment:
I'm not clear on why this is a `shared_ptr`, because it seems to me that it wouldn't really have to be shared with other classes. Are you planning to inject this in the constructor, or are you using it in some way that I'm not seeing?

Reply:
The bootstrap is also being used by the `bootstrap_task`, so it will still have to be a shared pointer in this case.